appservice/src/main/java/nu/yona/server/AppServiceApplication.java

```diff
@@ -200,4 +202,10 @@ private X509Certificate loadCertificateFromFile(String fileName, String descript
 			throw YonaException.unexpected(e);
 		}
 	}
+
+	@Bean
+	public LockPool<UUID> userSynchronizer()
```

Comment:
Each application (app service and analysis engine service) has their own lock pool bean, so these are unrelated. Theoretically, the analysis engine could run in a null pointer exception if it tries to update the user, but it can't do that as it only knows anonymized users.

You still have a valid point, for the other services. At this moment, I don't think any of them updates the service, but it's thinkable that one starts doing it. I actually favor implementing the right solution (through Hazelcast) rather than adding a lock pool for user locking to the other services too.

Reply:
OK, agreed.